### PR TITLE
Release 2.4.1

### PR DESCRIPTION
**Pull request by d4straub**
Release 2.4.1, mainly because of a nasty bug, see https://github.com/nf-core/ampliseq/pull/503. 
Other changes are relatively small, but bloated by template update 2.6.
For details see https://github.com/nf-core/ampliseq/blob/dev/CHANGELOG.md.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by erikrikarddaniel — APPROVED** [review]
LGTM.

File: assets/methods_description_template.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+id: "nf-core-ampliseq-methods-description"
+description: "Suggested text and references to use when describing pipeline usage within the methods section of a publication."
+section_name: "nf-core/ampliseq Methods Description"
+section_href: "https://github.com/nf-core/ampliseq"
+plot_type: "html"
+## TODO nf-core: Update the HTML below to your prefered methods description, e.g. add publication citation for this pipeline
```
Comment (mahesh-panchal) [inline assets/methods_description_template.yml:6]: Does this part need updating?

File: modules/local/picrust.nf [hunk 2]
```diff
@@ -18,7 +18,10 @@ process PICRUST {
     path("*_descrip.tsv"), emit: pathways
     path "versions.yml"  , emit: versions
     path "*.args.txt"    , emit: args
-    path "${message}.txt"
+    path "${message}.txt", emit: message
```
Comment (mahesh-panchal) [inline modules/local/picrust.nf:21]: missing `versions.yml`

File: modules/local/qiime2_diversity_core.nf [hunk 3]
```diff
@@ -17,6 +17,9 @@ process QIIME2_DIVERSITY_CORE {
     path "versions.yml"                         , emit: versions
     path("*rarefaction.txt")                    , emit: depth
 
+    when:
```
Comment (mahesh-panchal) [inline modules/local/qiime2_diversity_core.nf:20]: No `versions.yml` above.

File: subworkflows/local/qiime2_diversity.nf [hunk 4]
```diff
@@ -52,8 +52,10 @@ workflow QIIME2_DIVERSITY {
 
         //adonis ( ch_metadata, DIVERSITY_CORE.out.qza )
         if (params.qiime_adonis_formula) {
+            ch_qiime_adonis_formula = Channel.from(params.qiime_adonis_formula.tokenize(','))
```
Comment (mahesh-panchal) [inline subworkflows/local/qiime2_diversity.nf:55]:
```suggestion
            ch_qiime_adonis_formula = Channel.of(params.qiime_adonis_formula.tokenize(','))
```

File: modules/local/qiime2_featuretable_group.nf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        --i-table "filtered_${category}.qza" \
+        --p-axis 'sample' \
+        --m-metadata-file "${metadata}" \
+        --m-metadata-column "${category}" \
+        --p-mode 'sum' \
+        --o-grouped-table "${category}"
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        qiime2: \$( qiime --version | sed -e "s/q2cli version //g" | tr -d '`' | sed -e "s/Run qiime info for more version details.//g" )
```
Comment (mahesh-panchal) [inline modules/local/qiime2_featuretable_group.nf:38]:
```suggestion
        qiime2: \$( qiime --version | sed '1!d;s/.* //' )
```

File: modules/local/picrust.nf (hunk 2 above)
Comment (d4straub) [reply modules/local/picrust.nf:21]: versions.yml is there, line 19, but emit was missing, yes.

File: modules/local/qiime2_diversity_core.nf (hunk 3 above)
Comment (d4straub) [reply modules/local/qiime2_diversity_core.nf:20]: its in line 17, but I guess last line is sort of standard.

File: modules/local/qiime2_featuretable_group.nf (hunk 5 above)
Comment (d4straub) [reply modules/local/qiime2_featuretable_group.nf:38]: Thanks, that looks cleaner!

File: subworkflows/local/qiime2_diversity.nf (hunk 4 above)
Comment (d4straub) [reply subworkflows/local/qiime2_diversity.nf:55]: Thanks as well, .from seems to be deprecated, so I replace all by .of

File: modules/local/qiime2_featuretable_group.nf (hunk 5 above)
Comment (d4straub) [reply modules/local/qiime2_featuretable_group.nf]: Applied it, see below

File: subworkflows/local/qiime2_diversity.nf (hunk 4 above)
Comment (d4straub) [reply subworkflows/local/qiime2_diversity.nf]: Applied it, see below

File: assets/methods_description_template.yml (hunk 1 above)
Comment (d4straub) [reply assets/methods_description_template.yml]: Removed.

File: modules/local/qiime2_featuretable_group.nf [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    path "versions.yml" , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    """
+    export XDG_CONFIG_HOME="\${PWD}/HOME"
+
+    qiime feature-table filter-samples \
```
Comment (mahesh-panchal) [inline modules/local/qiime2_featuretable_group.nf:22]:
```suggestion
    qiime feature-table filter-samples \\
```
You need double slashes if you want to actually put line breaks in the code here.

File: modules/local/qiime2_featuretable_group.nf [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    conda (params.enable_conda ? { exit 1 "QIIME2 has no conda package" } : null)
+    container "quay.io/qiime2/core:2022.8"
+
+    input:
+    tuple path(table), path(metadata), val(category)
+
+    output:
+    path("${category}.qza")       , emit: qza
+    path "versions.yml" , emit: versions
```
Comment (mahesh-panchal) [inline modules/local/qiime2_featuretable_group.nf:13]:
```suggestion
    path("${category}.qza"), emit: qza
    path "versions.yml"    , emit: versions
```

**Review by mahesh-panchal — APPROVED** [review]
Couple of minor things, but otherwise LGTM

File: modules/local/qiime2_featuretable_group.nf (hunk 6 above)
Comment (mahesh-panchal) [reply modules/local/qiime2_featuretable_group.nf:22]: If you don't have the double slashes, the code prints on one line with a long space in between. The single `\` just adds a line break in the multi-line groovy string, but it's saying this should still be all on one line. The double `\\` put's a literal `\` in the shell code, which the single slash doesn't

**Comment by d4straub** [comment]
Thanks for the reviews!

**Pull request merged**